src/leaflike/db.clj

```diff
@@ -26,3 +26,22 @@
       (jdbc/insert! (db-spec) :bookmarks bkm)
       ; else
       {:error "Invalid Data"})))
+
+(defn list-all
+  []
+  (jdbc/query (db-spec) ["select * from bookmarks"]))
+
+
+(defn list-by-id
+  [params]
+  (jdbc/query (db-spec) ["select * from bookmarks
+                          where id = ?"
+                         (Integer/parseInt (:id params))]))
+
+(defn list-bookmark
+  [request]
+  (let [params (clojure.walk/keywordize-keys (-> request :query-params))]
+    (if (empty? params)
+      (list-all)
+      (if (= (is-valid-param? params) true)
```

Comment:
- `is-valid-param?` itself is a predicate. You don't need the equality check in the `if` block. It will work with the boolean that was returned by `is-valid-param?`

- Use `when` where you don't have an `else` case